tests/descriptors/topological.py

```diff
@@ -0,0 +1,164 @@
+import pytest
+from rdkit.Chem import MolFromSmiles, Mol, GetDistanceMatrix
+
+from skfp.descriptors import topological as top
+
+
+def diverse_molecules() -> list[tuple[str, Mol]]:
+    return [
+        ("ethane", MolFromSmiles("CC")),
+        ("ethanol", MolFromSmiles("CCO")),
+        ("carbon_dioxide", MolFromSmiles("O=C=O")),
+        ("benzene", MolFromSmiles("C1=CC=CC=C1")),
+        ("acetic_acid", MolFromSmiles("CC(=O)O")),
+        ("pyridine", MolFromSmiles("c1ccncc1")),
+        ("isobutane", MolFromSmiles("C(C)(C)C")),
+        ("pyrimidine", MolFromSmiles("c1cnc2ncnc12")),
+    ]
+
+
+@pytest.mark.parametrize(
+    "mol_name, expected_value",
+    [
+        ("ethane", 1),
+        ("ethanol", 4),
+        ("carbon_dioxide", 4),
+        ("benzene", 27),
+        ("acetic_acid", 9),
+        ("pyridine", 27),
+        ("isobutane", 9),
+        ("pyrimidine", 55),
+    ],
+)
+def test_weiner_index(mol_name, expected_value):
+    mol_tuple = next(item for item in diverse_molecules() if item[0] == mol_name)
+    mol = mol_tuple[1]
+
+    distance_matrix = GetDistanceMatrix(mol)
+
+    result = top.weiner_index(mol, distance_matrix)
+    result_without_distance_matrix = top.weiner_index(mol)
+
+    assert result == pytest.approx(expected_value, abs=1e-6)
+    assert result_without_distance_matrix == pytest.approx(expected_value, abs=1e-6)
+
+
+@pytest.mark.parametrize(
+    "mol_name, expected_value",
+    [
+        ("ethane", 1.0),
+        ("ethanol", 1.3333333333333333),
+        ("carbon_dioxide", 1.3333333333333333),
+        ("benzene", 1.8),
+        ("acetic_acid", 1.5),
+        ("pyridine", 1.8),
+        ("isobutane", 1.5),
+        ("pyrimidine", 1.9642857142857142),
```

Comment:
Round this to e.g. 3rd decimal place and then compare, it's more readable and still accurate enough